crates/rpc/rpc-eth-types/src/revm_utils.rs

```diff
@@ -136,12 +134,17 @@ impl CallFees {
                             RpcInvalidTransactionError::TipAboveFeeCap.into(),
                         )
                     }
-                    Ok(min(
-                        max_fee,
+
+                    let tip = max_fee.saturating_sub(block_base_fee);
+                    let effective_price = if tip > max_priority_fee_per_gas {
                         block_base_fee.checked_add(max_priority_fee_per_gas).ok_or_else(|| {
                             EthApiError::from(RpcInvalidTransactionError::TipVeryHigh)
-                        })?,
-                    ))
+                        })?
+                    } else {
+                        max_fee
```

Comment:
```suggestion
                        block_base_fee.checked_add(tip).ok_or_else(|| {
                            EthApiError::from(RpcInvalidTransactionError::TipVeryHigh)
```
based on the issue:

>  effective gas price is computed as:  priority_fee_per_gas + base_fee_per_gas,
where priority_fee_per_gas = min(tx.max_priority_fee_per_gas, tx.max_fee_per_gas - base_fee_per_gas)